url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/data/FileManager.java

```diff
@@ -0,0 +1,89 @@
+package darwInvest.data;
+
+import java.io.*;
+import java.net.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The FileManager class provides a convenient way
+ * to access and process files.
+ * Makes assumptions that everything works as expected,
+ * so it's not very safe in a real application.
+ * 
+ * @author Kevin Dolan
+ */
+public class FileManager {
+	
+	/**
+	 * Return a string representing the entire file
+	 * @param url the url to load
+	 * @return	  a string of the entire document contents
+	 */
+	public static String readWhole(String url) {
+		URL page = null;
+		try {
+			page = new URL(url);
+		} catch (MalformedURLException e1) {
+			e1.printStackTrace();
+		}
+		
+		String result = "";
+		try {
+			BufferedReader input = getReader(page);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *input*. The resource is referenced at line 36. The resource is closed at line 39. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *input* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).